dlt/destinations/impl/sqlalchemy/sqlalchemy_job_client.py

```diff
@@ -0,0 +1,503 @@
+    ) -> None:
+        self.capabilities = capabilities
+        self.dialect = dialect
+
+    def _db_integer_type(self, precision: Optional[int]) -> sa.types.TypeEngine:
+        if precision is None:
+            return sa.BigInteger()
+        elif precision <= 16:
+            return sa.SmallInteger()
+        elif precision <= 32:
+            return sa.Integer()
+        elif precision <= 64:
+            return sa.BigInteger()
+        raise TerminalValueError(f"Unsupported precision for integer type: {precision}")
+
+    def _create_date_time_type(
+        self, sc_t: str, precision: Optional[int], timezone: Optional[bool]
+    ) -> sa.types.TypeEngine:
+        """Use the dialect specific datetime/time type if possible since the generic type doesn't accept precision argument"""
+        precision = precision if precision is not None else self.capabilities.timestamp_precision
+        base_type: sa.types.TypeEngine
+        timezone = timezone is None or bool(timezone)
+        if sc_t == "timestamp":
+            base_type = sa.DateTime()
+            if self.dialect.name == "mysql":
+                # Special case, type_descriptor does not return the specifc datetime type
+                from sqlalchemy.dialects.mysql import DATETIME
+
+                return DATETIME(fsp=precision)
+        elif sc_t == "time":
+            base_type = sa.Time()
+
+        dialect_type = type(
+            self.dialect.type_descriptor(base_type)
+        )  # Get the dialect specific subtype
+        precision = precision if precision is not None else self.capabilities.timestamp_precision
+
+        # Find out whether the dialect type accepts precision or fsp argument
+        params = inspect.signature(dialect_type).parameters
+        kwargs: Dict[str, Any] = dict(timezone=timezone)
+        if "fsp" in params:
+            kwargs["fsp"] = precision  # MySQL uses fsp for fractional seconds
+        elif "precision" in params:
+            kwargs["precision"] = precision
+        return dialect_type(**kwargs)  # type: ignore[no-any-return,misc]
+
+    def _create_double_type(self) -> sa.types.TypeEngine:
+        if dbl := getattr(sa, "Double", None):
+            # Sqlalchemy 2 has generic double type
+            return dbl()  # type: ignore[no-any-return]
+        elif self.dialect.name == "mysql":
+            # MySQL has a specific double type
+            from sqlalchemy.dialects.mysql import DOUBLE
+        return sa.Float(precision=53)  # Otherwise use float
+
+    def _to_db_decimal_type(self, column: TColumnSchema) -> sa.types.TypeEngine:
+        precision, scale = column.get("precision"), column.get("scale")
+        if precision is None and scale is None:
+            precision, scale = self.capabilities.decimal_precision
+        return sa.Numeric(precision, scale)
+
+    def to_db_type(self, column: TColumnSchema, table_format: TTableSchema) -> sa.types.TypeEngine:
```

Comment:
there's PreparedTableSchema used in many places. this is a separate type on top of TTableSchema to make sure non-prepared tables do not leak to sql job client